recipes/bitserializer/all/conanfile.py

```diff
@@ -81,14 +103,41 @@ def source(self):
         get(self, **self.conan_data["sources"][self.version],
             destination=self.source_folder, strip_root=True)
 
+    def generate(self):
+        if not self._is_header_only:
+            tc = CMakeToolchain(self)
+            tc.variables["BUILD_CPPRESTJSON_ARCHIVE"] = self.options.with_cpprestsdk
+            tc.variables["BUILD_RAPIDJSON_ARCHIVE"] = self.options.with_rapidjson
+            tc.variables["BUILD_PUGIXML_ARCHIVE"] = self.options.with_pugixml
+            tc.variables["BUILD_RAPIDYAML_ARCHIVE"] = self.options.with_rapidyaml
+            tc.variables["BUILD_CSV_ARCHIVE"] = self.options.with_csv
+            tc.generate()
+            deps = CMakeDeps(self)
+            deps.generate()
+
     def build(self):
-        pass
+        if not self._is_header_only:
+            self._patch_sources()
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
 
     def package(self):
+        if not self._is_header_only:
+            cmake = CMake(self)
+            cmake.install()
+            rmdir(self, os.path.join(self.package_folder, "share"))
+        else:
+            copy(self, "*.h", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
+        # Copy license
         copy(self, "license.txt", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
-        copy(self, "*.h", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
+
+    def package_id(self):
+        if self._is_header_only:
+            self.info.header_only()
```

Comment:
```suggestion
            self.info.header_only()
        else:
            self.info.settings.clear()
```
Because you project does not need settings at all, as it only exports headers.
Also, does only CSV affect those exported headers?

Reply:
Options: `with_cpprestsdk`, `with_rapidjson`, `with_pugixml`, `with_rapidyaml` - produces only header files (not required to use CMake build)

Option: `with_csv` - produces header files and static library. This option requires settings for build.

I've added property `_is_header_only` which regulates when need to call CMake build and distinguish output packages by settings. Can't say that I expert of Conan, but as I undestand, for different settings should be produced different packages with unique hash which is calculated based on settings and options.

Reply:
Thank you for clarifying! I had a wrong vision of the artifacts produced by the project.